server/channels/api4/channel.go

```diff
@@ -1875,6 +1877,67 @@ func updateChannelScheme(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func getThreadsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId().RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), c.Params.ChannelId, model.PermissionReadChannel) {
+		c.SetPermissionError(model.PermissionReadChannel)
+		return
+	}
+
+	opts := model.GetChannelThreadsOpts{
+		PageSize: uint64(c.Params.PerPage),
+	}
+
+	queryValues := r.URL.Query()
+
+	opts.Deleted, _ = strconv.ParseBool(queryValues.Get("deleted"))
+	opts.Since, _ = strconv.ParseUint(queryValues.Get("since"), 10, 64)
+	opts.Before = queryValues.Get("before")
+	opts.After = queryValues.Get("after")
+	opts.Extended, _ = strconv.ParseBool(queryValues.Get("extended"))
+	opts.TotalsOnly, _ = strconv.ParseBool(queryValues.Get("totals_only"))
+	opts.ThreadsOnly, _ = strconv.ParseBool(queryValues.Get("threads_only"))
```

Comment:
@hmhealey 

- `deleted`: mobile "specific", we need to know which were deleted to remove them from 🍉 
- `since`: sync threads on web-socket re-connection, not yet used.
- `before`, `after`: pagination
- `exteded`: (?)
- `totalsOnly`, `threadsOnly`: those we are added to the other thread APIs so we don't need to calculate counts when paginating, however it seems that we don't use `totalsOnly` on this API.

We could remove from the client `since` and `totalsOnly`, but to remove them from the server we'll need to see mobile app's needs first.